fire/test_main.py

```diff
@@ -0,0 +1,25 @@
+"""
+Test using fire via python -m fire
+"""
+import os
+
+from fire import __main__
+from fire.core import FireExit
+from fire import testutils
+
+class TestAsMainModule(testutils.BaseTestCase):
+  def test_name_setting(self):
+    # confirm one of the usage lines has tempfile gettempdir
+    # with self.assertRaisesFireExit(2, 'tempfile gettempdir'):
+    with self.assertOutputMatches('tempfile gettempdir'):
```

Comment:
I'm confused by the behavior here - argparse raises `SystemExit(0,)` if you say `-h` and `SystemExit(2,)` if not enough arguments, so I don't understand here (where it's printing out a usage statement) why it doesn't exit either 0 or 2 at this point.

(I think we've discussed this before too, but maybe I'm forgetting).

Commented out line is how I *think* it should work, but what actually works is uncommented.

Reply:
It's exiting with SystemExit(0), but not with a FireExit. Here are the current rules:

- Whenever Fire encounters a FireError (this occurs when when a Fire command cannot be executed, and is different from user code raising an exception), Fire raises a FireExit with code 2 (Recall that FireExit is a subclass of SystemExit, and so the program exists).
- When Fire is used with the help or trace flags, Fire raises a FireExit with code 0 if successful.
- Otherwise (the default case), the program just ends, which is equivalent to a SystemExit(0), but not a FireExit

Reply:
gotcha - I think I was expecting it to set a code of 2 just like argparse, but pretty bike-sheedy difference